### PR TITLE
Improve the integration tests framework documentation

### PR DESCRIPTION
**Pull request by belimawr**
## What does this PR do?

It improves the integration tests framework documentation on how to debug test from the remove VMs.

## Why is it important?

It improves the integration tests framework documentation on how to debug test from the remove VMs.

## Checklist

- ~~[ ] My code follows the style guidelines of this project~~
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- [x] I have made corresponding changes to the documentation
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~~
- ~~[ ] I have added an integration test or an E2E test~~

~~## Disruptive User Impact~~
~~## How to test this PR locally~~
~~## Related issues~~

## Questions to ask yourself

- How are we going to support this in production?
- How are we going to measure its adoption?
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

File: docs/test-framework-dev-guide.md [hunk 1]
```diff
@@ -136,28 +136,70 @@ when running them manually, such as `ELASTICSEARCH_HOST`, `ELASTICSEARCH_USERNAM
...
-machines, the best way to circumvent that is to debug the tests
-directly on a VM. `mage integration:DeployDebugTools` will show a menu
-to select a VM and then install the common debugging tools: Delve,
-Mage and Docker. It will also create the `~/elastic-agent` folder
-containing the Git repository (required to package from within the VM)
+#### Connecting to VMs
+All VMs (including Windows) support connections via SSH, the framework
+generates and stores the necessary SSH keys to access the VMs, the
+easiest way to connect to them is using the SSH command returned by
+`mage integration:SSHVM`. It will list the VMs and ask to select
```
Comment (mauri870) [inline docs/test-framework-dev-guide.md:143]: I'm following the onboarding documentation and noticed that `integration:SSHVM` is not a valid mage target. The correct one seems to be `integration:ssh`. Could you please update it? Thanks.

Comment (belimawr) [reply docs/test-framework-dev-guide.md]: Thank you for spotting that! I fixed it on [a4a2f16](https://github.com/elastic/elastic-agent/pull/5461/commits/a4a2f16739af41196072f3c895ea292ed097a95f)

**Review by swiatekm — APPROVED** [review]
Looks good as far as I can tell, but I'd prefer someone more familiar with the framework to review too.

**Review by michel-laterman — APPROVED** [review]

**Comment by belimawr** [comment]
It seems this PR is waiting on SonarQube even though it only changes documentation. :thinking: 

I enabled auto-merge and I'll look more into it tomorrow.

**Comment by ycombinator** [comment]
No need for SonarQube check on this docs-only PR.  Force merging...

**Pull request merged**